crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -176,6 +178,29 @@ impl Solution {
         Settlement::encode(self, auction, eth, simulator).await
     }
 
+    pub fn with_protocol_fees(self) -> Result<Self, fee::Error> {
+        let mut trades = Vec::new();
```

Comment:
nit: To communicate to the reader that we know upfront how many trades to expect.
I assume it was done this way because propagating errors out of `.map()` is a bit annoying.
```suggestion
        let mut trades = Vec::with_capacity(self.trades.len());
```

Reply:
adjusted.